client/src/client.rs

```diff
@@ -571,50 +539,40 @@ impl<DB: Database> Client<DB> {
         block: BlockTag,
     ) -> Result<U256> {
         self.node
-            .read()
-            .await
+            .load_full()
             .get_storage_at(address, slot, block)
             .await
     }
 
     pub async fn send_raw_transaction(&self, bytes: &[u8]) -> Result<H256> {
-        self.node.read().await.send_raw_transaction(bytes).await
+        self.node.load_full().send_raw_transaction(bytes).await
     }
 
     pub async fn get_transaction_receipt(
         &self,
         tx_hash: &H256,
     ) -> Result<Option<TransactionReceipt>> {
-        self.node
-            .read()
-            .await
-            .get_transaction_receipt(tx_hash)
-            .await
+        self.node.load_full().get_transaction_receipt(tx_hash).await
     }
 
     pub async fn get_transaction_by_hash(&self, tx_hash: &H256) -> Result<Option<Transaction>> {
-        self.node
-            .read()
-            .await
-            .get_transaction_by_hash(tx_hash)
-            .await
+        self.node.load_full().get_transaction_by_hash(tx_hash).await
     }
 
     pub async fn get_logs(&self, filter: &Filter) -> Result<Vec<Log>> {
-        self.node.read().await.get_logs(filter).await
+        self.node.load_full().get_logs(filter).await
     }
 
-    pub async fn get_gas_price(&self) -> Result<U256> {
-        self.node.read().await.get_gas_price()
+    pub fn get_gas_price(&self) -> Result<U256> {
+        self.node.load_full().get_gas_price()
```

Comment:
could we use `.load()` in every place that we don't `.await`? From what I understood, `.load_full()` makes a deep copy and is useful for when the value will be held for a longer time. Here we just read last blocks data and return

Reply:
Yes, I missed this one. `load_full` does a deep-copy of `Arc`, not the `Node`, so memory usage is not expensive. It is just more expensive than `load`. Also, `load` can do `load_full` transparently if needed, check [this](https://docs.rs/arc-swap/1.6.0/arc_swap/strategy/type.DefaultStrategy.html).